src/app/qgisapp.cpp

```diff
@@ -6492,18 +6481,15 @@ bool QgisApp::addProject( const QString &projectFile )
 #ifdef WITH_BINDINGS
     if ( mPythonUtils && mPythonUtils->isEnabled() )
     {
-      // does the project have any macros?
-      if ( !QgsProject::instance()->readEntry( QStringLiteral( "Macros" ), QStringLiteral( "/pythonCode" ), QString() ).isEmpty() )
+      const bool projectContainsMacros = !QgsProject::instance()->readEntry( QStringLiteral( "Macros" ), QStringLiteral( "/pythonCode" ), QString() ).isEmpty();
+      const bool projectContainsExpressionFunctions = !QgsProject::instance()->readEntry( QStringLiteral( "ExpressionFunctions" ), QStringLiteral( "/pythonCode" ), QString() ).isEmpty();
+      if ( projectContainsMacros || projectContainsExpressionFunctions )
```

Comment:
isn't this an incomplete check? what about actions and edit forms? I think this should be using the QgsProjectUtils method instead

Reply:
@nyalldawson , the idea here is to defer the trust request until action / attribute form triggers / needs it.

Reply:
Ok, fair enough